go-controller/pkg/ovn/policy.go

```diff
@@ -1116,21 +1156,27 @@ func (oc *Controller) handleLocalPodSelectorDelFunc(np *networkPolicy, objs ...i
 	return nil
 }
 
-func (oc *Controller) addLocalPodHandler(policy *knet.NetworkPolicy, np *networkPolicy,
-	handleInitialItems func([]interface{}) error) error {
-
+// This function starts a watcher for local pods. Sync function and add event for every existing pod
+// will be executed sequentially first, and an error will be returned if something fails.
+// LocalPodSelectorType uses handleLocalPodSelectorAddFunc on Add and Update,
+// and handleLocalPodSelectorDelFunc on Delete. These functions wil return nil when networkPolicy.deleted
+// is set to true
+func (oc *Controller) addLocalPodHandler(policy *knet.NetworkPolicy, np *networkPolicy) error {
 	// NetworkPolicy is validated by the apiserver
 	sel, err := metav1.LabelSelectorAsSelector(&policy.Spec.PodSelector)
 	if err != nil {
 		klog.Errorf("Could not set up watcher for local pods: %v", err)
 		return err
 	}
 
+	syncFunc := func(objs []interface{}) error {
+		return oc.handleLocalPodSelectorAddFunc(np, true, objs...)
```

Comment:
ignoreErr initially means ignore individual object errros, do best-effort.
but now I am thinking, since error from syncFunction will cause networkPolicy creation failure, shouldn't we just ignore all errors returned from `handleLocalPodSelectorAddFunc` on handler create? And then rely on per-pod events to retry
I updated the code with that idea for now, lmk what you think

Reply:
yeah I think this makes sense. Only concern would be if the pods weren't added during sync, then they might get added to default deny before they would be allowed by the policy port group. I think this is a nuanced failure case though and we can just accept that behavior for now.

Reply:
local ports are added to networkPolicy and default deny port groups with the same transaction, so allow and deny acls should be applied (almost) at the same time